### PR TITLE
Refactor/partitioning worker

### LINKED ISSUE #1277: partitioning thread worker code is duplicated across two files
from `do-partition.py` and `partition-graph.py`

### PR DESCRIPTION
**Pull request by mr-c**
Fixes #1277

### CODE REVIEW

**Comment by mr-c** [comment]
- [ ] Is it mergeable?
- [ ] Did it pass the tests?
- [ ] If it introduces new functionality in scripts/ is it tested?
  Check for code coverage with `make clean diff-cover`
- [ ] Is it well formatted? Look at `make pep8`, `make diff_pylint_report`,
  `make cppcheck`, and `make doc` output. Use `make format` and manual
  fixing as needed.
- [ ] Did it change the command-line interface? Only additions are allowed
  without a major version increment. Changing file formats also requires a
  major version number increment.
- [ ] Is it documented in the ChangeLog?
  http://en.wikipedia.org/wiki/Changelog#Format
- [ ] Was a spellchecker run on the source code and documentation after
  changes were made?
- [ ] Do the changes respect streaming IO? (Are they
  tested for streaming IO?)
- [ ] Is the Copyright year up to date?

**Comment by mr-c** [comment]
@camillescott can I get a pre-review?

File: oxli/partition.py [hunk 1]
```diff
@@ -0,0 +1,51 @@
+#! /usr/bin/env python
+#
+# This file is part of khmer, http://github.com/ged-lab/khmer/, and is
+# Copyright (C) Michigan State University, 2009-2015. It is licensed under
+# the three-clause BSD license; see doc/LICENSE.txt.
```
Comment (camillescott) [inline oxli/partition.py]: We're updating this right?

File: oxli/partition.py [hunk 2]
```diff
@@ -0,0 +1,51 @@
...
+import os.path
+
+# stdlib queue module was renamed on Python 3
+try:
+    import queue
+except ImportError:
+    import Queue as queue
+
+
+def worker(que, basename, stop_big_traversals):
```
Comment (camillescott) [inline oxli/partition.py]: *squints*

**Comment by camillescott** [comment]
Other than the copyright and that var name, LGTM

**Comment by camillescott** [comment]
Although, now that it's been separated into its own func, you should write a test for it :)

File: oxli/partition.py (hunk 1 above)
Comment (mr-c) [inline oxli/partition.py]: In a separate PR, and for the whole codebase, yes/

File: oxli/partition.py (hunk 2 above)
Comment (mr-c) [inline oxli/partition.py]: ha! Forgot, sorry; this was from a previous branch.

File: oxli/partition.py [hunk 3]
```diff
@@ -0,0 +1,51 @@
...
+        try:
+            (nodegraph, index, start, stop) = que.get(False)
+        except queue.Empty:
+            print('exiting', file=sys.stderr)
+            return
+
+        outfile = basename + '.subset.%d.pmap' % (index,)
+        if os.path.exists(outfile):
+            print('SKIPPING', outfile, ' -- already exists', file=sys.stderr)
+            continue
```
Comment (mr-c) [inline oxli/partition.py]: these two lines have no test coverage (why do we continue instead of stopping?)

**Comment by mr-c** [comment]
This has been rebased & tested; ready for merge @ctb

**Comment by ctb** [comment]
LGTM - thanks!

**Pull request merged**